subsys/net/ip/tcp.c

```diff
@@ -545,6 +545,9 @@ static void tcp_send_timer_cancel(struct tcp *conn)
 	}
 
 	k_work_cancel_delayable(&conn->send_timer);
+#if defined(CONFIG_NET_TCP_KEEPALIVE)
+	k_work_cancel_delayable(&conn->keepalive_timer);
+#endif /*CONFIG_NET_TCP_KEEPALIVE */
```

Comment:
style nit, this looks better if `/* CONFIG....`, please do search and replace for the whole file